src/main/java/com/scottlogic/sorts/KeywordSearch.java

```diff
@@ -0,0 +1,54 @@
+package com.scottlogic.sorts;
+
+import com.scottlogic.UserPost;
+import com.scottlogic.filters.ContentFilter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class KeywordSearch {
+
+    public List<UserPost> keywordSearchSort(List<UserPost> inputList, String keyWord) {
+
+        ContentFilter filterByKeyWord = new ContentFilter(keyWord);
+
+        List<UserPost> listToBeSorted = filterByKeyWord.filter(inputList);
+        int[] count = new int[listToBeSorted.size()];
+        for (int i = 0; i < listToBeSorted.size(); i++) {
+            String content = listToBeSorted.get(i).getContents();
+            count[i] = 0;
+            String[] words = content.split(" ");
+
+            for (int j = 0; j < words.length; j++) {
+                if (words[j].compareToIgnoreCase(keyWord) == 0) {
+                    count[i]++;
+                }
+            }
+        }
+        int tempOrder;
+        int tempCount;
+        int[] order = new int[count.length];
+        for (int i = 0; i < order.length; i++) {
+            order[i] = i;
+        }
+
+        for (int i = 0; i < listToBeSorted.size(); i++) {
```

Comment:
I'm not too clear what's going on here, but it looks like the actual sorting. It's a bit complicated - do you think it could be simplified at all?